# Patch for heartbeat to support drbd versions >=8.3
# downloaded from http://hg.linux-ha.org/dev/rev/6467be4d4cb7
# to apply: cd /usr/share/ocf/resource.d/heartbeat/; patch -p3 < <patch file>
#
# --- original patch ---
# HG changeset patch
# User Dejan Muhamedagic <dejan@hello-penguin.com>
# Date 1242229157 -7200
# Node ID 6467be4d4cb72698fcb345ec7bf6440450abc4fb
# Parent  34eea1564ffda85ca34bbd4945b8e9c07ab5c6bc
# Medium: RA: drbd: support drbd versions >=8.3
--- a/resources/OCF/drbd	Wed May 13 17:37:42 2009 +0200
+++ b/resources/OCF/drbd	Wed May 13 17:39:17 2009 +0200
@@ -232,9 +232,38 @@
 	return 1
 }
 
-
+get_drbd_ver() {
+	# returns 2 for unsupported (not 0.7.x or < 8.x)
+	# returns 1 for versions prior to 8.3.x
+	# returns 0 otherwise
+	drbdadm  | grep Version | awk '{print $2}' |
+	awk -F. '
+		{
+			if( $1 == "0" ) # 0.7.x
+				{ v1 = $2; v2 = $3; }
+			else
+				{ v1 = $1; v2 = $2; }
+		}
+		v1 < 7 { exit 2; }
+		v1 == 7 || (v1 == 8 && v2 < 3) { exit 1; } # use state
+		# otherwise use role
+	'
+}
+get_status_cmd() {
+	get_drbd_ver
+	rc=$?
+	if [ $rc -ge 2 ]; then
+		ocf_log err "Cannot parse output of 'drbdadm | grep Version'"
+		exit $OCF_ERR_GENERIC 
+	elif [ $rc -eq 1 ]; then
+		echo state
+	else
+		echo role
+	fi
+}
 drbd_get_status() {
-	DRBD_STATE=$(do_drbdadm state $RESOURCE)
+	cmd=`get_status_cmd`
+	DRBD_STATE=$(do_drbdadm "$cmd" $RESOURCE)
 	DRBD_STATE_LOCAL=$(echo $DRBD_STATE | sed -e 's#/.*##')
 	DRBD_STATE_REMOTE=$(echo $DRBD_STATE | sed -e 's#.*/##')
 	DRBD_CSTATE=$(do_drbdadm cstate $RESOURCE)

